src/Symfony/Bundle/FrameworkBundle/Routing/AnnotatedRouteControllerLoader.php

```diff
@@ -43,14 +43,12 @@ protected function configureRoute(Route $route, \ReflectionClass $class, \Reflec
      */
     protected function getDefaultRouteName(\ReflectionClass $class, \ReflectionMethod $method)
     {
-        return preg_replace([
-            '/(bundle|controller)_/',
-            '/action(_\d+)?$/',
-            '/__/',
-        ], [
-            '_',
-            '\\1',
-            '_',
-        ], parent::getDefaultRouteName($class, $method));
+        $name = preg_replace('/(bundle|controller)_/', '_', parent::getDefaultRouteName($class, $method));
+
+        if ('Action' === substr($method->name, -6) || '_action' === substr($method->name, -7)) {
```

Comment:
IMO we should require the php 8 polyfill and use str_ends_with here. Ref. #41576

Reply:
I've opened a PR to do this in the whole codebase: #41973